src/nsrt_learning.py

```diff
@@ -95,20 +109,28 @@ def learn_strips_operators(segments: Sequence[Segment], verbose: bool = True,
     """
     # Partition the segments according to common effects.
     params: List[Sequence[Variable]] = []
+    option_vars: List[Tuple[Variable, ...]] = []
     add_effects: List[Set[LiftedAtom]] = []
     delete_effects: List[Set[LiftedAtom]] = []
     partitions: List[Partition] = []
     for segment in segments:
+        if segment.has_option():
+            segment_option_objs = tuple(segment.get_option().objects)
+        else:
+            segment_option_objs = tuple()
         for i in range(len(partitions)):
             # Try to unify this transition with existing effects.
             # Note that both add and delete effects must unify.
+            part_option_vars = option_vars[i]
             part_add_effects = add_effects[i]
             part_delete_effects = delete_effects[i]
             suc, sub = unify_effects_and_options(
```

Comment:
Good thing we forgot to rename this method earlier! Now it's the correct name again, lol

Reply:
Should we get rid of the default argument values for ground_option_args and lifted_option_args, in the `def unify_effects_and_options`